src/utils/hooks/useRouter.tsx

```diff
@@ -64,6 +65,10 @@ export function useRouter(): RouterResult {
           delete newQueryObject[key]
         } else newQueryObject[key] = value
       })
+
+      // Prevents un-necessary re-renders when query hasn't actually changed
+      if (isEqual(newQueryObject, queryFilters)) return
+
```

Comment:
This was a big score! I suspect this has been the cause of many unnecessary re-renders throughout the app.

I could figure out why the `useDataViews` hooks were making so many repeated requests even when the `useEffect` dependencies weren't changing.

Turns out, whenever we run `updateQuery`, it creates a new instance of the `query` object. And React detects changes by *reference*, not *value*, so it was causing re-renders even when the query values didn't change (which happens a lot in the loading phase of many components).

We now do a deep-equality check on the updated query object, and if it hasn't actually changed values, we exit without updating the history or query state in the Router.

Sooo much smoother!